### DISCUSSION TITLE
tileserver-gl alternative?

### DISCUSSION THREAD

**Discussion by koufopoulosf** [maplibre/martin#647] [Q&A]
Hello,

I just found out this wonderful repo. I would love to ask, how does this compare to [tileserver-gl](https://github.com/maptiler/tileserver-gl) ?

This one renders the tiles faster than tileserver-gl? Is there any benchmark with the above mentioned tile server?

Also, does it come with any handy pre-made styles like the tileserver-gl does?

Thanks in advance for your time!

Best Regards.

**Comment by nyurik** [comment]
@koufopoulosf we should have a comparison table (PR are welcome ;)).  In short, tileserve-gl is PROBABLY slower due to javascript vs Rust, but this has not been explicitly tested. More importantly, tileserve-gl focuses on serving just from mbtiles, whereas Martin can also serve directly from a postgres server(s) - tables or functions, and also from pmtiles. Plus it can combine multiple tile sources on the fly, so you can get a single tile that actually comes from several backends.  One area where Martin is behind is server-side rendering - if you want your server to return a PNG that it dynamically creates from an internal MVT, than you need server-side rendering which Martin doesn't yet support

**Comment by koufopoulosf** [comment]
@nyurik thank you very much for your detailed response. Personally, one key difference I noticed is that tilesserver-gl makes it a lot easier to create beautiful basemaps like this: https://maps.ckochis.com/styles/basic-preview/?vector

Could you please give me a hint on what I should do to achieve such result using martin ? Can we serve such basemap using martin eitger with mbtiles or pmtiles?

How do I even access the tiles after the command "./martin tiles.pmtiles" or "./martim tiles.mbtiles"?

Best Regards.

**Comment by nyurik** [comment]
@koufopoulosf that is a good question - we do need a better "quick start" documentation. Martin focuses on one task: retrieve a tile (from one of the backends) and send it to the user. Martin does not (yet) handle styling aspects like serving fonts, sprites, SDFs, styling sheet, javascript assets, etc.  In other words, it kinda assumes you will just have a static server with all those things somewhere, compiled from maplibre + fonts + ..., and that static service will simply point to martin to get the actual map tiles.

**Comment by koufopoulosf** [comment]
@nyurik Thank you very much for your response. I was kind of lost with it. So the best option we have so far to serve customized style (beautiful) vector tile basemaps is using tileserver-gl and .mbtiles? Is there any other option maybe you would suggest as a better alternative?

**Comment by nyurik** [comment]
not exactly - if you can put together all the needed static web assets (maplibre gl js + style + sprites + fonts) into a single dir, you can serve it directly with nginx - much simpler than having the less stable custom service like tileserve-gl. And that code can request tiles from Martin service because it should offer faster performance, more stability, and more data sources (e.g. you can generate your tiles on the fly from some data tables in postgres).

**Comment by koufopoulosf** [comment]
I really like your idea but I cannot find resources to implement such solution. I was not even able to proceed or do anything after the "./martin tiles.mbtiles" command. If you could please give any hints or even better, make a tutorial on such solution, I really believe it would allow the martin project to become more popular. But for now, I really cannot implement anything with it, unfortunately.

If you could make such tutorial, I would appreciate it, a lot.

Best Regards.

**Comment by nyurik** [comment]
i posted it in [our slack](https://osmus.slack.com/archives/C01G3D28DAB/p1682360444387049) - feel free to discuss it there, or propose this as a bounty for maplibre.

**Comment by koufopoulosf** [comment]
It requires an @openstreetnap.us account to join slack without invitation.

**Comment by nyurik** [comment]
its automated - see https://github.com/maplibre/maplibre-gl-js#getting-involved

**Comment by koufopoulosf** [comment]
Got it, thank you! Where can I see the discussion? I have access only to random and general channel. Thanks once again!

**Comment by nyurik** [comment]
search for maplibre channels. You have access to all of them
